src/test/ripple-libpp_demo.cpp

```diff
@@ -178,7 +171,189 @@ int main (int argc, char** argv)
     auto const allPass = std::all_of(passes.begin(), passes.end(),
         [](auto p) { return p; });
     assert(allPass);
-    std::cout << (allPass ? "All checks pass.\n" : "Some checks fail.\n");
+    std::cout << (allPass ?
+        "All single signing checks pass.\n" :
+        "Some single signing checks fail.\n");
+
+    return allPass;
+}
+
+//------------------------------------------------------------------------------
+
+// Demonstrate multisigning.
+
+// Helper function that throws if for some reason we can't create a seed.
+ripple::Seed getSeed (std::string const& seedText)
+{
+    // WARNING!
+    // Never use ripple::parseGenericSeed() for secure code.  Call
+    // ripple::randomSeed() instead, since it is cryptographically secure.
+    boost::optional<ripple::Seed> const possibleSeed {
+        ripple::parseGenericSeed (seedText)};
+
+    if (! possibleSeed)
+        throw std::runtime_error (
+            std::string ("Could not generate seed from: ") + seedText);
+
+    return *possibleSeed;
+}
+
+// Holds identifying information for an account.
+class Credentials
+{
+    std::string const name_;
+    ripple::KeyType const keyType_;
+    ripple::Seed const seed_;
+    std::pair<ripple::PublicKey, ripple::SecretKey> const keys_;
+    ripple::AccountID const id_;
```

Comment:
After thrashing a bit I ended up using `generateKeyPair()` because it gives the same answers as `wallet_propose`.  (I've found `wallet_propose` to be a really useful tool when testing stuff with rippled.)  But I wanted the `Credentials` members to be `const`, and it would be really wasteful to call `generateKeyPair()` twice.  So I stored `PublicKey` and `SecretKey` in a `std::pair<>`, because that is how they are returned by `generateKeyPair()`.

Then, since I was storing a `std::pair`, I wanted to give users of `Credentials` a better name than `keys.first` for the PublicKey.  Hence getters.